app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -60,6 +61,18 @@ public void addAll(List<Show> list) {
         notifyDataSetChanged();
     }
 
+    class NavigateFeedToDetail implements NavigationInterface {
+        public void navigate(View v, Bundle b){
+            Navigation.findNavController(v).navigate(R.id.action_feedFragment_to_showDetailFragment, b);
+        }
+    }
+
+    class NavigateSavedToDetail implements NavigationInterface {
+        public void navigate(View v, Bundle b){
+            Navigation.findNavController(v).navigate(R.id.action_savedFragment_to_showDetailFragment, b);
+        }
+    }
```

Comment:
These implementations should live in the classes where you need it (so the fragments where you setup the adapters).  Or I would pull it out into a navigation module.  They shouldn't really live in the adapter.  The adapter doesn't need to know about all of the implementations